staging/src/k8s.io/apiserver/pkg/storage/etcd3/feature/feature_support_checker_test.go

```diff
@@ -0,0 +1,147 @@
+limitations under the License.
+*/
+
+package feature
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+// MockEtcdClient is a mock implementation of the EtcdClientInterface interface.
+type MockEtcdClient struct {
+	Version string
+	// Indicates whether to return an error on Status and message of that error.
+	Error error
+}
+
+func (m *MockEtcdClient) Endpoints() []string {
+	return []string{"localhost:2390"}
+}
+
+// Status returns a mock status response.
+func (m *MockEtcdClient) Status(ctx context.Context, endpoint string) (*clientv3.StatusResponse, error) {
+	if m.Error != nil {
+		return nil, m.Error
+	}
+	// Return a mock status response
+	return &clientv3.StatusResponse{
+		Version: m.Version,
+	}, nil
+}
+
+func TestCheckSupportedRequestWatchProgress(t *testing.T) {
+	tests := []struct {
+		testName              string
+		mockClientVersion     string
+		mockStatusReturnError error
+		expectedSupportResult bool
+		expectedError         error
+	}{
+		{
+			testName:              "Supported version, bound",
+			mockClientVersion:     "3.4.31",
+			expectedSupportResult: true,
+		},
+		{
+			testName:              "Supported version, bound",
+			mockClientVersion:     "3.5.13",
+			mockStatusReturnError: nil,
+			expectedSupportResult: true,
+			expectedError:         nil,
+		},
+		{
+			testName:              "Supported version",
+			mockClientVersion:     "3.6.0",
+			expectedSupportResult: true,
+		},
+		{
+			testName:              "Unsupported version",
+			mockClientVersion:     "3.4.30",
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Unsupported version",
+			mockClientVersion:     "3.5.0",
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Unsupported version",
+			mockClientVersion:     "3.5.12",
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Mock client status return error",
+			mockClientVersion:     "3.5.13",
+			mockStatusReturnError: fmt.Errorf("Error from status"),
+			expectedSupportResult: false,
+			expectedError:         fmt.Errorf("failed checking etcd version, endpoint: %q: %w", "localhost:2390", errors.New("Error from status")),
+		},
+		{
+			testName:              "Malformed version, error of malformed version logged via klog",
+			mockClientVersion:     "3.5.--a",
+			expectedSupportResult: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			// Mock Etcd client
+			mockClient := &MockEtcdClient{Version: tt.mockClientVersion, Error: tt.mockStatusReturnError}
+
+			ctx := context.Background()
+			supported, err := endpointSupportsRequestWatchProgress(ctx, mockClient, "localhost:2390")
+
+			// Assertions
+			assert.Equal(t, tt.expectedSupportResult, supported)
+			assert.Equal(t, tt.expectedError, err)
+		})
+	}
+}
+
+func TestSupports(t *testing.T) {
```

Comment:
We could expand the tests to cover couple of scenarios:
* Default disabled
* Supports if provided client has at least one endpoint that supports it and no client that doesn't.
* Disables if called with client doesn't support it

etc

Reply:
Thanks, I'll add.